packages/datadog-instrumentations/src/jest.js

```diff
@@ -0,0 +1,143 @@
+'use strict'
+
+const { addHook, channel, AsyncResource } = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+const testStartCh = channel('ci:jest:test:start')
+const testSkippedCh = channel('ci:jest:test:skip')
+const testRunEndCh = channel('ci:jest:test:end')
+const testErrCh = channel('ci:jest:test:err')
+const testSuiteEnd = channel('ci:jest:test-suite:end')
+
+const {
+  getTestSuitePath,
+  getTestParametersString
+} = require('../../dd-trace/src/plugins/util/test')
+
+const { getFormattedJestTestParameters } = require('../../datadog-plugin-jest/src/util')
+
+const specStatusToTestStatus = {
+  'pending': 'skip',
+  'disabled': 'skip',
+  'todo': 'skip',
+  'passed': 'pass',
+  'failed': 'fail'
+}
+
+function getWrappedEnvironment (NodeEnvironment) {
+  return class DatadogEnvironment extends NodeEnvironment {
+    constructor (config, context) {
+      super(config, context)
+      this.testSuite = getTestSuitePath(context.testPath, config.rootDir)
+      this.ar = new AsyncResource('bound-anonymous-fn')
+      this.nameToParams = {}
+    }
+    async teardown () {
+      super.teardown().finally(() => {
+        testSuiteEnd.publish()
+      })
+    }
+
+    async handleTestEvent (event, state) {
+      if (super.handleTestEvent) {
+        await super.handleTestEvent(event, state)
+      }
+      this.ar.runInAsyncScope(() => {
+        let context
+        if (this.getVmContext) {
+          context = this.getVmContext()
+        } else {
+          context = this.context
+        }
+
+        const setNameToParams = (name, params) => { this.nameToParams[name] = params }
+
+        if (event.name === 'setup') {
```

Comment:
this handler is called once with `event.name === 'setup'`. At this moment we use the chance to wrap `global.test.each`, which is used for parameterized tests